onnx/backend/test/cmd_tools.py

```diff
@@ -46,17 +46,31 @@ def prepare_dir(path):  # type: (Text) -> None
                     output_dir, 'test_data_set_{}'.format(i))
                 prepare_dir(data_set_dir)
                 for j, input_np in enumerate(inputs):
-                    tensor = numpy_helper.from_array(
-                        input_np, case.model.graph.input[j].name)
+                    if isinstance(input_np, dict):
+                        arr = numpy_helper.from_dict_to_map(
+                            input_np, case.model.graph.input[j].name)
+                    elif input_np.dtype == 'object':
```

Comment:
this seems to be picking `test_strnorm_model_monday_casesensintive_nochangecase` as sequence as well. The input for the above test case is `array(['monday', 'tuesday', 'wednesday', 'thursday'], dtype=object)`.

Reply:
Ah, yes, just noticed strings are treated as dtype=object in np. Will try to identify ideas for how to better specify sequences.

Reply:
Why not use python dictionaries and lists as the representation for ONNX's map and sequence? Won't that be more natural in Python?

Reply:
I had initially implemented it with dicts and lists, but ran into a significant number of errors in the testing infrastructure as everything was tailored to np.array formats.

Reply:
However, I do think it is the more natural data structure within Python. I'll look into it a bit more.

Reply:
See recent updates.